apache/zeppelin
# [ZEPPELIN-1711] Create `Dockerfile`s for released bin

### What is this PR for?

Created `Dockerfile` for released bin

- based on **Ubuntu:16.04 (LTS)** for desktop usage
- **JDK 8**
- **R** with basic packages
- **Python 2** with basic packages
- **miniconda2** for `%python.conda`

### Details

We already discussed about using alpine image in https://github.com/apache/zeppelin/pull/1761.

- However, it's not designed for desktop usage  
- Doesn't have some official packages (R, ...)
- Not familiar to users for desktop OS

That the reason why ubuntu is used in base image

```
zeppelin                  base                b3818f9ae4b1        11 hours ago        1.67 GB
zeppelin                  0.6.2               c0a4d8556f92        7 hours ago         2.29 GB
zeppelin                  0.7.0               c4a5ad0d04bd        8 hours ago         2.5 GB
zeppelin                  0.7.1               54173b77743b        7 hours ago         2.49 GB
```

### What type of PR is it?
[Feature]

### Todos
* [x] - base image
* [x] - script for creating bin images
* [x] - bin image template

### What is the Jira issue?

[ZEPPELIN-1711](https://issues.apache.org/jira/browse/ZEPPELIN-1711)

### How should this be tested?

1. build base image `cd scripts/docker/zeppelin/base; docker build -t zeppelin:base ./`
2. build bin image `cd scripts/docker/zeppelin/0.7.1; docker build -t zeppelin:0.7.1 ./`
3. execute docker images 

```
docker run -p 8080:8080 --rm --name zeppelin zeppelin:0.7.1
```

since it takes time to build, you can use already [published docker images](https://hub.docker.com/r/1ambda/docker-zeppelin/)

```
docker run -p 8080:8080 --rm --name zeppelin 1ambda/docker-zeppelin:0.7.1
```

4. should be able to run spark, python and R tutorials

### Screenshots (if appropriate)

NO

### Questions:
* Does the licenses files need update? - NO
* Is there breaking changes for older versions? - NO
* Does this needs documentation? - YES, updated
